repo/config/profile.go

```diff
@@ -24,54 +33,96 @@ var defaultServerFilters = []string{
-		c.Discovery.MDNS.Enabled = false
-		return nil
+var Profiles = map[string]Profile{
+	"server": {
+		Description: `Disables local host discovery, recommended when
+running IPFS on machines with public IPv4 addresses.`,
+
+		Transform: func(c *Config) error {
+			c.Addresses.NoAnnounce = appendSingle(c.Addresses.NoAnnounce, defaultServerFilters)
+			c.Swarm.AddrFilters = appendSingle(c.Swarm.AddrFilters, defaultServerFilters)
+			c.Discovery.MDNS.Enabled = false
+			return nil
+		},
 	},
-	"local-discovery": func(c *Config) error {
-		c.Addresses.NoAnnounce = deleteEntries(c.Addresses.NoAnnounce, defaultServerFilters)
-		c.Swarm.AddrFilters = deleteEntries(c.Swarm.AddrFilters, defaultServerFilters)
-		c.Discovery.MDNS.Enabled = true
-		return nil
+
+	"local-discovery": {
+		Description: `Sets default values to fields affected by the server
+profile, enables discovery in local networks.`,
+
+		Transform: func(c *Config) error {
+			c.Addresses.NoAnnounce = deleteEntries(c.Addresses.NoAnnounce, defaultServerFilters)
+			c.Swarm.AddrFilters = deleteEntries(c.Swarm.AddrFilters, defaultServerFilters)
+			c.Discovery.MDNS.Enabled = true
+			return nil
+		},
 	},
-	"test": func(c *Config) error {
-		c.Addresses.API = "/ip4/127.0.0.1/tcp/0"
-		c.Addresses.Gateway = "/ip4/127.0.0.1/tcp/0"
-		c.Addresses.Swarm = []string{
-			"/ip4/127.0.0.1/tcp/0",
-		}
-
-		c.Swarm.DisableNatPortMap = true
-
-		c.Bootstrap = []string{}
-		c.Discovery.MDNS.Enabled = false
-		return nil
+	"test": {
+		Description: `Reduces external interference of IPFS daemon, this
+is useful when using the daemon in test environments.`,
+
+		Transform: func(c *Config) error {
+			c.Addresses.API = "/ip4/127.0.0.1/tcp/0"
+			c.Addresses.Gateway = "/ip4/127.0.0.1/tcp/0"
+			c.Addresses.Swarm = []string{
+				"/ip4/127.0.0.1/tcp/0",
+			}
+
+			c.Swarm.DisableNatPortMap = true
+
+			c.Bootstrap = []string{}
+			c.Discovery.MDNS.Enabled = false
+			return nil
+		},
 	},
-	"default-networking": func(c *Config) error {
-		c.Addresses = addressesConfig()
+	"default-networking": {
+		Description: `Restores default network settings.
+Inverse profile of the test profile.`,
+
+		Transform: func(c *Config) error {
+			c.Addresses = addressesConfig()
 
-		c.Swarm.DisableNatPortMap = false
-		c.Discovery.MDNS.Enabled = true
-		return nil
+			c.Swarm.DisableNatPortMap = false
+			c.Discovery.MDNS.Enabled = true
+			return nil
+		},
 	},
-	"badgerds": func(c *Config) error {
-		c.Datastore.Spec = map[string]interface{}{
-			"type":   "measure",
-			"prefix": "badger.datastore",
-			"child": map[string]interface{}{
-				"type":       "badgerds",
-				"path":       "badgerds",
-				"syncWrites": true,
-			},
-		}
-		return nil
+	"badgerds": {
+		Description: `Replaces default datastore configuration with experimental
+badger datastore.
+
+If you apply this profile after ipfs init, you will need
+to convert your datastore to the new configuration.
+You can do this using ipfs-ds-convert.
```

Comment:
Can we put a usage example of ipfs-ds-convert as well? Or maybe a link to the readme of ipfs-ds-convert is enough

Reply:
Done